pkg/broker/handler/handler.go

```diff
@@ -45,62 +47,108 @@ type Handler struct {
 	// to 1.
 	Concurrency int
 
+	ConfigCh chan *config.Broker
+
 	// cancel is function to stop pulling messages.
 	cancel context.CancelFunc
+
+	done <-chan struct{}
 }
 
 // Start starts the handler.
 // done func will be called if the pubsub inbound is closed.
-func (h *Handler) Start(ctx context.Context, done func(error)) {
+func (h *Handler) Start(ctx context.Context, config *config.Broker) {
```

Comment:
Currently, there is a 1-1 correspondence between brokers and handlers, and context is used as a form of implicit argument passing from handlers to processors and as well as between processors. The [context docs](https://golang.org/pkg/context/#example_WithValue) warn against this:
> Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions.

I think in this case if we need a handler for a different purpose, we should simply write a new handler rather than trying to write an all-purpose handler as this level of abstraction is not well suited for go.

Reply:
I see it as a suggestion rather than requirement. Knative code in a lot of places doesn't follow this suggestion - client/informer injection, logger in the context, etc.

Reply:
Knative is definitely guilty of abusing context as well, but I don't necessarily think we should copy that practice if we can avoid it.